File: sol/solution/0001-0100/0019. Remove Nth Node From End of List/Solution.dart
/**
 * Definition for singly-linked list.
 * class ListNode {
 *   int val;
 *   ListNode? next;
 *   ListNode([this.val = 0, this.next]);
 * }
 */
class Solution {
  ListNode? removeNthFromEnd(ListNode? head, int n) {
    ListNode dummy = ListNode(0, head);
    ListNode? first = dummy;
    ListNode second = dummy;
    for (int i = 0; i <= n; i++) {
      first = first?.next;
    }
    while (first != null) {
      first = first.next;
      second = second.next!;
    }
    second.next = second.next?.next;
    return dummy.next;
  }
}

File: sol/solution/0001-0100/0019. Remove Nth Node From End of List/Solution.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @param {Integer} n
# @return {ListNode}
def remove_nth_from_end(head, n)
    dummy = ListNode.new(0, head)
    first = dummy
    second = dummy
    n.times { first = first.next }
    while first.next
        first = first.next
        second = second.next
    end
    second.next = second.next.next
    dummy.next
end